### ISSUE TITLE
Holoviews does not install reliably with pip

### ISSUE THREAD

**Issue by jordansamuels** [ioam/holoviews#1734]
It appears that holoviews does not properly install with pip in a clean environment, and that
the last release that installed properly is 1.7.0 .  See further down on clean reproduction.

I've tested this on Centos 7 / Python 3.6.0.  Installing 1.8.0 or 1.8.1 with or without `[all]`
results in an error message about params which is similar to the one supplied in a comment to
https://github.com/ioam/holoviews/issues/1659 . Sometimes the error is about numpy instead of params.

These commands fail:
* `pip install holoviews`
* `pip install holoviews[all]`
* `pip install holoviews==1.8.0`
* `pip install holoviews[all]==1.8.0`

These commands succeed:
* `pip install holoviews==1.7.0`
* `pip install holoviews[all]==1.7.0`

## Clean reproduction
Because issue is difficult to test and reproduce reliably, I have reproduced it using
[Docker](http://www.docker.com). I get the same behavior whether I'm using conda or pyenv.
Steps to reproduce cleanly:

1. [Install Docker](https://docs.docker.com/engine/installation/)
2. Use this [Dockerfile](https://gist.github.com/jordansamuels/f8ec860c7b49034e9be338cb20c58337) for conda, or this [Dockerfile](https://gist.github.com/jordansamuels/27ef6440a7df0256b0f6c437db531a79) for pyenv.  Put them in a `conda` or `pyenv` directory and `cd` to it.
3. `docker build -t repro .` 
4. `docker run -t repro pip install holoviews` or another similar command from above.

**Comment by jordansamuels** [comment]
Sample failure:
```
% docker run -ti hv-issue-pyenv  pip install 'holoviews[all]'
Collecting holoviews[all]
  Downloading holoviews-1.8.1.tar.gz (3.1MB)
    100% |################################| 3.1MB 64kB/s
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-build-f6dq5rx3/holoviews/setup.py", line 131, in <module>
        package_assets(example_path)
      File "/tmp/pip-build-f6dq5rx3/holoviews/setup.py", line 117, in package_assets
        import holoviews
      File "/tmp/pip-build-f6dq5rx3/holoviews/holoviews/__init__.py", line 5, in <module>
        import numpy as np # noqa (API import)
    ModuleNotFoundError: No module named 'numpy'
```

**Comment by jordansamuels** [comment]
If we manually do `pip install numpy param` first, then `pip install holoviews` will work.  However, this approach isn't available with automated environment management, e.g. if we have `requirements.txt` as 

```
numpy
param
holoviews
```

Then `pip install -r requirements.txt` still fails in the same way as noted in the main issue.

**Comment by tdsmith** [comment]
The proximal cause is that setup.py imports holoviews, which tries to import holoview's dependencies before setuptools has had a chance to inspect or install the dependencies. https://github.com/ioam/holoviews/blob/master/setup.py#L117

**Comment by jlstevens** [comment]
I think that is correct. Thanks for diagnosing this!

Recently setup.py was refactored which is why we are now encountering these problems with pip. One solution might be to simply copy the code from ``holoviews.util.examples`` into ``setup.py`` although that will mean an annoying duplication of code.

**Comment by jlstevens** [comment]
This should be fixed in the PR reference above (#1782). Closing.
